bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/handler/EnOceanBaseActuatorHandler.java

```diff
@@ -90,14 +94,30 @@ protected EnOceanActuatorConfig getConfiguration() {
         return (EnOceanActuatorConfig) config;
     }
 
+    @Override
+    Collection<EEPType> getEEPTypes() {
+        Collection<EEPType> r = super.getEEPTypes();
+        if (sendingEEPType == null) {
+            return r;
+        }
+        
+        return Collections.unmodifiableCollection(Stream.concat(r.stream(), Collections.singletonList(sendingEEPType).stream()).collect(Collectors.toList()));
+    }
+
     @Override
     boolean validateConfig() {
+        
+        try {
+            sendingEEPType = EEPType.getType(getConfiguration().sendingEEPId);
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
I must confess that I was a little bit to lazy to catch all different errors (config could not be parsed, SendingEEP not provided or SendingEEP not valid). I check all this and return a better error description now.